src/format/scan.rs

```diff
@@ -198,9 +198,67 @@ pub(super) fn space(s: &str) -> ParseResult<&str> {
     }
 }
 
-/// Consumes any number (including zero) of colon or spaces.
+/// Returns slice remaining after first char.
+/// If <=1 chars in `s` then return an empty slice
+pub(super) fn s_next(s: &str) -> &str {
+    match s.char_indices().nth(1) {
+        Some((offset, _)) => &s[offset..],
+        None => {
+            // one or zero chars in `s`, return empty string
+            &s[s.len()..]
+        }
+    }
+}
+
+/// If the first `char` is whitespace then consume it and return `s`.
+/// Else return `s`.
+pub(super) fn space1(s: &str) -> &str {
+    match s.chars().next() {
+        Some(c) if c.is_whitespace() => s_next(s),
+        Some(_) | None => s,
+    }
+}
+
+/// Allow a colon with possible one-character whitespace padding.
+/// Consumes zero or one of these leading patterns:
+/// `":"`, `" "`, `" :"`, `": "`, or `" : "`.
 pub(super) fn colon_or_space(s: &str) -> ParseResult<&str> {
-    Ok(s.trim_left_matches(|c: char| c == ':' || c.is_whitespace()))
+    let c0_ = match s.chars().next() {
```

Comment:
Okay, can we just get rid of the underscore suffixes? I don't see their value.

Reply:
Fixed.